T7-T9: Landing Page Body, Nav Bar, and Footer

hferoz1 opened the pull request:
I created the nav bar, footer, and landing page body in one js file. If necessary, I can separate them. In the future, I will create more detailed pull requests!

hferoz1 committed:
Landing Page for Career Connect

justinpchen94 commented on LandingPage.css:
```diff
@@ -0,0 +1,97 @@
+  line-height: 1.6;
+}
+
+.navBar {
```
in general css prefers hyphens rather than camelcase so `nav-bar` instead of `navBar`

justinpchen94 commented on LandingPage.css:
you're going to want to rename this to `LandingPage.module.css` in order to ensure that your CSS changes are contained to just the components importing this CSS file!

justinpchen94 commented on LandingPage.js:
let's move this file into `frontend/src/components/LandingPage/LandingPage.js` ideally and similarly for the CSS file above

fahadakhan2 replied:
Looks like this still needs to be done @hferoz1

justinpchen94 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+    );
+}
+
+export default LandingPage;
```
nit: add newline to end of file

justinpchen94 reviewed (commented):
looks pretty clean! leaving a few comments but let's also remember to always include an image in the PR description whenever you are changing or building something UI related. this helps reviewers a ton in terms of figuring out what your code changes look like

hferoz1 committed:
Update LandingPage.module.css

hferoz1 committed:
Update and rename LandingPage.css to LandingPage.module.css

hferoz1 committed:
Update LandingPage.js

hferoz1 commented:
Updated PR: 
- changed the name of the CSS file 
- updated use of classnames 
- added a new line after the last code line the js file
- some styling changed for CSS

fahadakhan2 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+import styles from './LandingPage.module.css';
+import classNames from 'classnames';
```
not sure what this import is, but it seems like its not being used so you can remove this

fahadakhan2 commented on LandingPage.js:
You will most likely use the nav bar and footer on many parts of your app, not just the landing page. So it is better to make them into separate components and create them in their own files and utilize them throughout the app where needed.

fahadakhan2 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+        <div>
+            <header>
+                <div className={styles.navBar}>
+                    <a href="#home">HOME</a>
```
Instead of a tags for your routes to different parts of your application you should use Link from React Router. You will have to npm install react router dom first before you are able to utilize it throughout your app.

https://reactrouter.com/en/main/components/link

fahadakhan2 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+                    <div className={styles.content}>
+                        <h2>Loren Ipsum</h2>
+                        <p>
+                            Lorem ipsum dolor sit amet, sed unum adversarium disputationi eu, aliquid conceptam mel ut. Vel modo nusquam ad, cu sea oratio legere ornatus. Quo ex ignota senserit, ei sea soluta fuisset. Sea ex impetus mediocrem intellegat. Sea ad enim nobis convenire, his brute velit ea, mel an clita reprimique.
```
Since you're taking on the task of creating the landing page we should try to implement to its finished product. This looks like you put temporary text for now but please change this to the actual content for the landing page of career connect. 

Here is the mockup for the landing page that Jason created for us to have a idea of how the landing page may look but you can get as creative as you want: https://www.figma.com/design/TzYtjBdMGoKkkG2w1QyMwV/Career-Roadmap-%26-Skill-Swap?node-id=9-53&t=S0IXMADczwRD8ynl-0

hferoz1 replied:
Can I change this once we start working more on the different features? I think we're trying to making it less like LinkedIn and I want to make sure I'm explaining the site's use properly

fahadakhan2 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+                        </p>
+                    </div>
+                    <div className={styles.content}>
+                        <img className={styles.img3} src="IMGblank.png" alt="Placeholder" />
```
for your css selectors be as descriptive as possible. For example instead of img3 name it what the image actually is.

fahadakhan2 commented on LandingPage.js:
```diff
@@ -0,0 +1,69 @@
+                    <img className={styles.img5} src="IMGblank.png" alt="Placeholder" />
+                </section>
+            </div>
+            <footer>
```
Take a look at the wireframes to see how the footer can be filled out with more info: https://www.figma.com/design/TzYtjBdMGoKkkG2w1QyMwV/Career-Roadmap-%26-Skill-Swap?node-id=9-53&t=S0IXMADczwRD8ynl-0

fahadakhan2 commented on LandingPage.module.css:
```diff
@@ -0,0 +1,87 @@
+.navBar {
```
change the css names to kebab case so .nav-bar

hferoz1 committed:
Delete frontend directory

hferoz1 committed:
Delete LandingPage.module.css

hferoz1 committed:
Delete LandingPage.js

hferoz1 committed:
Updated this PR

Not sure how to add this to the a the proper folder without starting a new pull request.
In this commit, I have:
-  changed everything from camelcase to kebab case
- removed unnecessary import statements 
- put the landing page, nav bar, and footer in separate js files
- created a better footer

hferoz1 commented:
![img1](https://github.com/Your-Missing-Semester/purple_repo/assets/157250037/2c0f75f3-5148-4739-a378-abdd32b1aefe)
![img2](https://github.com/Your-Missing-Semester/purple_repo/assets/157250037/fbda90d6-0c2c-4894-8e37-17d00c984303)
![img3](https://github.com/Your-Missing-Semester/purple_repo/assets/157250037/d326c7a0-63c4-43bd-900e-c4a880c83dc8)

A few screenshots of the updated UI!

justinpchen94 commented on LandingPage/Footer.js:
looks like this file and the `LandingPage` directory itself were moved to the wrong directory level? i believe it should be nested as `frontend/src/components/LandingPage`

hferoz1 replied:
i'm having trouble moving this into the right level, is it okay if I open a new pr?

justinpchen94 commented on frontend/src/App.css:
were these files deleted on purpose? looks like we might need to do some git surgery hahah you may be able to use https://www.freecodecamp.org/news/git-checkout-file-from-another-branch/ to pull files from the `main` branch and restore these

hferoz1 replied:
I deleted the files, and then uploaded the new versions of the files if that's okay

justinpchen94 reviewed (commented):
not sure if this is a preemptive review but it looks like you might have deleted some files unnecessarily
